### PR TITLE
feat: port Description from console

### PR DESCRIPTION
**Pull request by chambo-e**

### CODE REVIEW

File: src/components/Description/index.tsx [hunk 1]
```diff
@@ -1,56 +1,70 @@
...
+        overflow: hidden;
+        text-overflow: ellipsis;
+        width: ${width};
+      }
+    `}
 
-  ${({ theme, selectable }) =>
-    selectable &&
+  ${({ userSelect, theme }) =>
+    userSelect &&
```
Comment (matthprost) [inline src/components/Description/index.tsx:65]: Use ternary instead ?

```ts
ellipsis ? ... : null
```

File: src/components/Description/index.tsx [hunk 2]
```diff
@@ -1,56 +1,70 @@
...
+  ${({ inline }) =>
+    inline &&
+    css`
+      display: flex;
+      justify-content: flex-start;
+      align-items: center;
+      ${Desc} {
+        margin: 0 0 0 8px;
+      }
+    `};
```
Comment (matthprost) [inline src/components/Description/index.tsx:51]: Here too

File: src/components/Description/index.tsx [hunk 3]
```diff
@@ -1,56 +1,70 @@
...
+
+const Desc = styled('dd', {
+  shouldForwardProp: prop => !['capitalize'].includes(prop),
+})<{ capitalize?: boolean }>`
+  display: flex;
+  align-items: center;
   color: ${({ theme }) => theme.colors.neutral.text};
-  margin: 0;
-  margin-top: ${({ theme }) => theme.space['1']};
+  margin: 8px 0 0 0;
```
Comment (matthprost) [inline src/components/Description/index.tsx:19]: Can we use theme spacing ?

File: src/components/Description/index.tsx [hunk 4]
```diff
@@ -1,56 +1,70 @@
...
+  userSelect?: boolean
+  width?: string
+}
+
+const BareDescription = styled('dl', {
+  shouldForwardProp: prop =>
+    !['inline', 'ellipsis', 'userSelect'].includes(prop),
+})<DescriptionProps>`
   font-size: 16px;
   line-height: 16px;
```
Comment (matthprost) [inline src/components/Description/index.tsx:39]: Possible to use text / theme.typography properties instead of hard coded values ?

**Review by matthprost — CHANGES_REQUESTED** [review]

File: src/components/Description/index.tsx [hunk 5]
```diff
@@ -1,56 +1,70 @@
 import { css } from '@emotion/react'
 import styled from '@emotion/styled'
-import PropTypes from 'prop-types'
-import { WeakValidationMap } from 'react'
-import Box, { BoxProps } from '../Box'
 
-const Term = styled(Box.withComponent('dt'))`
+const Term = styled.dt`
   font-weight: 500;
```
Comment (matthprost) [inline src/components/Description/index.tsx:5]: Same here we could use values from a Text or use `theme.typography`

**Review by matthprost — CHANGES_REQUESTED** [review]

File: src/components/Description/index.tsx (hunk 1 above)
Comment (chambo-e) [reply src/components/Description/index.tsx:65]: @matthprost Why ? In template literal it does not makes a lot of sense 🤔

File: src/components/Description/index.tsx (hunk 4 above)
Comment (chambo-e) [reply src/components/Description/index.tsx:39]: There is no typography with a line-height 16

Comment (matthprost) [reply src/components/Description/index.tsx:39]: Maybe we should see with design what kind of typography we can use in this case?

File: src/components/Description/index.tsx (hunk 1 above)
Comment (matthprost) [reply src/components/Description/index.tsx:65]: @chambo-e why not?

Comment (chambo-e) [reply src/components/Description/index.tsx:65]: Sure, it's just more verbose with no safety addition :)

File: src/components/Description/index.tsx (hunk 4 above)
Comment (chambo-e) [reply src/components/Description/index.tsx:39]: @matthprost so process design system, can we leave it as is for this PR and refactor it when the component as been spec-ed ?

Comment (matthprost) [reply src/components/Description/index.tsx:39]: @chambo-e yes go ahead :)

**Review by matthprost — APPROVED** [review]

**Review by philibea — APPROVED** [review]

**Pull request merged**